crux-llvm/test/Test.hs

```diff
@@ -124,8 +143,10 @@ getZ3Version =
   let getVer (Right inp) =
         -- example inp: "Z3 version 4.8.7 - 64 bit"
         let w = words inp
-        in if and [ length w > 2, head w == "Z3" ]
-           then w !! 2 else "?"
+        in if |  ("Z3":_:verNum:_) <- w
```

Comment:
I suppose these could also have avoided the warnings by changing `head w` to `w !! 0`, but the pattern matching approach is fine as well.  I'd suggest further simplifying by replacing the `let w = words inp in` with `if | [ ... ] <- words inp`, but it's up to you.

Reply:
Good point. In fact, we can get rid of the use of `MultiWayIf` and just use plain-old mattern matching here, which makes the code even simpler.